### ISSUE TITLE
Tilde character ~ is not displayed in terminal using Belgian keyboard layout

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#174]
_From @pierredewilde on July 9, 2016 18:30_
- VSCode Version: 1.3.0
- OS Version: Mac OS X El Capitan 10.11.5 with Belgian keyboard

Steps to Reproduce:
1. Open Integrated Terminal (`View > Integrated Terminal`)
2. Type `cd ~` + Enter

The tilde character (~) is not displayed.
`$ cd`

_Copied from original issue: Microsoft/vscode#9008_

**Comment by Tyriar** [comment]
- Does it show up if you hit `~` multiple times?
- Does it actually input a `~` or skip the character (does the cursor move right when you hit `~`, what does tab completion do when you type `cd ~/`?

**Comment by Tyriar** [comment]
_From @pierredewilde on July 10, 2016 4:1_

- The tilde does not appear even if hit multiple times.
- The cursor does not move. `cd ~/` is inputed as `cd /`

It is a keyboard layout issue: if I switch to the US keyboard layout, I can type `~`. On a Belgian keyboard layout, I need to type `alt+n` followed by a space to produce `~`. In the code editor, `~` appears. In the integrated terminal, `~` does not appear.

If I copy `cd ~` from the code editor and paste it in the integrated terminal, it works as expected.

**Comment by Tyriar** [comment]
_From @nih0 on July 11, 2016 8:52_

I have the same issue with a german keyboard on macOS(`OS X 10.11.1`).

- `OS X 10.11.1`
- `Version 1.3.0`

**Comment by Tyriar** [comment]
Thanks, I think this is to do with lack of AltGr support (#7061). Please confirm :smiley:

**Comment by Tyriar** [comment]
_From @pierredewilde on July 11, 2016 21:28_

I am using a MacBookAir with a Belgian keyboard layout:
https://support.apple.com/library/content/dam/edam/applecare/images/en_US/keyboards/belgian_keyboard.png

There is no AltGr key on Apple keyboard layout.

**Comment by Tyriar** [comment]
@pierredewilde if possible, could you setup https://github.com/sourcelair/xterm.js and check to see if this reproduces in the demo? This should get it up and running easily:

```bash
git clone https://github.com/sourcelair/xterm.js.git
cd xterm.js
npm install
npm start # I think this needs node <= 4
# open browser at http://0.0.0.0:3000
```

**Comment by Tyriar** [comment]
_From @pierredewilde on July 11, 2016 23:21_

Yep, same issue after installing and running `xterm.js`: no tilde appears.

BTW, you don't need `node <= 4`. Tested with `node 6.3.0`.

**Comment by Tyriar** [comment]
I can reproduce on Ubuntu, `~~` types a single `~` and `~[space]` types a single `~`.

**Comment by Tyriar** [comment]
@pierredewilde this also happens in the other programs I tried it out in? (Chromium and gedit)

**Comment by pierredewilde** [comment]
It is not the case on Mac with international keyboard: `~~` types nothing and `~[space]`types a space. `~` never appears in integrated terminal.

This issue is specific to integrated terminal. It works as expected anywhere else (Terminal, Chromium, ...).

**Comment by Tyriar** [comment]
I think this was fixed with improved alt support in 3.2.0 https://github.com/xtermjs/xterm.js/releases/tag/3.2.0

**Comment by marco-2017** [comment]
Working on Windows with an Azerty keyboard, typing Shift + F4 creates the tilde straight away, for me.
